tests/envs/test_register.py

```diff
@@ -143,7 +143,7 @@ def test_env_version_suggestions(
 def test_register_versioned_unversioned():
     # Register versioned then unversioned
     versioned_env = "Test/MyEnv-v0"
-    gym.register(versioned_env)
+    gym.register(versioned_env, None)
```

Comment:
Is this being type checked? Because you're simultaneously disallowing `entry_point=None` in the type system, and adding an explicit `entry_point=None` in the usage.

Reply:
Agreed, I was confused about that as well. I think it is because we disabled `GeneralTypeIssues="none"`.  I will change it to "no-entry-points"

Reply:
If you want more explicit semantic in the type system:
```python
from enum import Enum, auto
class GymEnum(Enum):
    no_entry_points = auto()

...
entry_point: Union[Callable, str, Literal[GymEnum.no_entry_points]]
```

Reply:
This is purely for testing so I think that is unnecessary and could cause more issues for users